chapter-6/exercises.pl

```diff
@@ -100,37 +100,13 @@
 %% identical to List2, except that the first and last elements are
 %% exchanged. Hint: here's where append comes in useful again.
 
-%% Helper function
-
-%% equal
-%% base case
-equal([],[]).
-%% inductive case
-equal([H1|T1], [H1|T2]) :- equal(T1,T2).
-
-%% reverse
-%% base case
-accRev([],A,A).
-%% inductive case
-accRev([H|T],A,R) :- accRev(T,[H|A],R).
-%% main
-rev(L,R) :- accRev(L,[],R).
-
-chopOffEnds([_|T],Outlist) :-
-  rev(T, [_|RevOutlist]),
-  rev(RevOutlist,Outlist).
-
-%% Actual function
-
-%% swapfl(List1, List2)
-swapfl(List1,List2) :-
-  compareFirstLast(List1,List2),
-  compareFirstLast(List2,List1),
-  chopOffEnds(List1, Outlist1),
-  chopOffEnds(List2, Outlist2),
-  equal(Outlist1,Outlist2).
-
-compareFirstLast([H|T1],L2) :- rev(L2, [H|_]).
+%% As per strict interpretation of the question. Empty lists or lists with one
+%% element cannot be compared since they are missing the first and last element.
```

Comment:
The first and last elements of the list with length 1 `[a]` are both `a`, so the predicate `swapfl([a], [a])` should be true, at least the way I read the task's description. The case with zero length lists is not so clear though.
P.s. Bare in mind that if you add a `swapfl([X], [X])` clause, the second rule will start to (wrongly) accept the case with 2 equal 2-element lists (`swapfl([a, b], [a, b]).` will be true).